### ISSUE TITLE
Error for import type, `Imported multiple times` in typescript

### ISSUE THREAD

**Issue by njfamirm** [import-js/eslint-plugin-import#2843]
Hello
On the `2.28.0` version of this package, I'm getting an error from eslint: `Imported multiple times`.
This happens when I use import type in addition to normal import!

For example:
``Ts
import {createServer} from 'node:http';
import type {IncomingMessage, ServerResponse} from 'node:http';
```
In the above code, we have the same error from [eslint-plugin-import](https://github.com/import-js/eslint-plugin-import)

This is the complete dependabot lint workflow [log](https://github.com/AliMD/alwatr/actions/runs/5691651979/job/15427232981)

**Comment by Tirke** [comment]
Impacting us quite a lot, reverted for the moment.

**Comment by unicornware** [comment]
@njfamirm given your example, passing code for this rule would be the following:

```ts
import {
  createServer,
  type IncomingMessage,
  type ServerResponse
} from 'node:http'
```

**Comment by unicornware** [comment]
there **is** an error with this rule, however, when **_all_** type imports are from the same file:

```ts
import type testSubject from '../properties'
import type { Properties } from '../properties'
```

yields an `imported multiple times` error.

**Comment by Tirke** [comment]
> @njfamirm given your example, passing code for this rule would be the following:
> 
> ```ts
> import {
>   createServer,
>   type IncomingMessage,
>   type ServerResponse
> } from 'node:http'
> ```
> 
> (assuming the rule in question is `no-duplicates` 😅)

Yes we can make it pass by doing that but as type imports will not end up in the outputted JS, it would be good if we can actually differentiate (like the previous version did) between purely type imports and imports when it comes to `no-duplicate`.

**Comment by njfamirm** [comment]
> @njfamirm given your example, passing code for this rule would be the following:
> 
> ```ts
> import {
>   createServer,
>   type IncomingMessage,
>   type ServerResponse
> } from 'node:http'
> ```
> 
> (assuming the rule in question is `no-duplicates` 😅)

Yes, you're right.
My question is, is this a bug? or a new rule of this plugin?
If it's a rule, we should decide to disable them or solve lint errors in our code.

**Comment by ljharb** [comment]
`no-duplicates` just looks at the specifier. It makes sense to me that if there's only one `import` and only one `import type` from the same specifier, that it would not warn - I'd be happy to review a PR that made that change.

**Comment by askoufis** [comment]
Does https://github.com/import-js/eslint-plugin-import/pull/2716 address this issue?

I've also run into a similar situation in the following case with `prefer-inline: true`:
```tsx
import type React from 'react';
import { useEffect } from 'react';
```

The rule complains about duplicate imports, but the fix merges the imports incorrectly anyway. This is one case that should not be fixed IMO, though `import { useEffect, type default as React } from 'react';` is a valid fix.

**Comment by ljharb** [comment]
Yes, I believe #2716 will address that.

**Comment by njfamirm** [comment]
> > @njfamirm given your example, passing code for this rule would be the following:
> > ```ts
> > import {
> >   createServer,
> >   type IncomingMessage,
> >   type ServerResponse
> > } from 'node:http'
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > (assuming the rule in question is `no-duplicates` 😅)
> 
> Yes, you're right. My question is, is this a bug? or a new rule of this plugin? If it's a rule, we should decide to disable them or solve lint errors in our code.

ok, this is new rule, so we update our codes to remove this lint error.
Thank you and everyone who answered ❤️
